File: src/etch/bindings/csharp/msg/Validator_ArrayValue.cs
using System;
using System.Collections.Generic;
using System.Text;
using Etch.Transport;

namespace Etch.Msg
{
    /// <summary>
    /// Validator for ArrayValue
    /// </summary>
    public class Validator_ArrayValue : TypeValidator
    {
        public static Validator_ArrayValue Get()
        {
            if ( validator == null )
                validator = new Validator_ArrayValue();

            return validator;
        }

        private static Validator_ArrayValue validator;

        private Validator_ArrayValue() : base( typeof( ArrayValue ), typeof( ArrayValue ), 0, "ArrayValue" )
        {
        }

        public override Validator ElementValidator()
        {
            return null;
        }      
    }
}

File: src/etch/bindings/java/msg/Validator_ArrayValue.java
/* $Id$
 *
 * Created by wert on Aug 27, 2007.
 *
 * Copyright (c) 2007 Cisco Systems, Inc. All rights reserved.
 */

package etch.bindings.java.msg;

import etch.bindings.java.transport.fmt.TypeCode;

/**
 * Validator for ArrayValue.
 */
public class Validator_ArrayValue extends TypeValidator
{
	/**
	 * @return an instance of the validator.
	 */
	public static Validator_ArrayValue get()
	{
		if (validator == null)
			validator = new Validator_ArrayValue();
		
		return validator;
	}
	
	private static Validator_ArrayValue validator;
	
	/**
	 * Constructs the Validator.
	 */
	private Validator_ArrayValue()
	{
		super( ArrayValue.class, ArrayValue.class, 0, "ArrayValue" );
	}

	@Override
	public Validator elementValidator()
	{
		return null;
	}
	
	@Override
	public Byte checkValue( Object value )
	{
		if (!validate( value))
			return null;
		
		return TypeCode.ARRAY;
	}
}
